[feat] 옷장 상세 조회 API 구현

zerone1202 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 --> 

- Closes #22 
- 옷장의 상세 정보를 조회할 수 있는 API를 구현했습니다.


## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- closetId를 PathVariable로 받아 해당 옷장의 상세 정보 조회
- 인증된 사용자만 접근 가능 
- 존재하지 않거나 삭제된 옷장에 대한 404 예외 처리
- Spring Security가 인증/인가를 자동으로 처리하므로 403 에러코드는 별도로 정의하지 않음
- 삭제된 옷장(isDeleted = true)도 존재하지 않는 것으로 간주하여 404 반환
- @Transactional(readOnly = true)로 조회 성능 최적화
- record의 기본 생성자 활용으로 불필요한 Builder 제거

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)## 💬 공유사항 to 리뷰어<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->
- 옷장 상세 조회 성공
<img width="1249" height="722" alt="스크린샷 2025-10-19 21 36 12" src="https://github.com/user-attachments/assets/794e9263-3393-4703-b462-5face1d465c1" />

- 존재하지 않는 옷장 조회 시
<img width="1249" height="567" alt="스크린샷 2025-10-19 21 36 34" src="https://github.com/user-attachments/assets/bd14ba52-3d3b-4899-afe1-009ec908f803" />

- 인증 안된 사용자 접근 시 403 에러 발생 확인
<img width="1249" height="485" alt="스크린샷 2025-10-19 20 57 37" src="https://github.com/user-attachments/assets/826ac70e-5841-4dad-8757-340b5c9280f5" />

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

zerone1202 committed:
[feat] closet: ClosetGetResponse DTO 생성

- 옷장 상세 조회 응답 DTO 추가
- record 타입으로 불변 객체 구현
- 정적 팩토리 메서드 from()으로 Entity → DTO 변환
- Builder 대신 record 기본 생성자 활용

zerone1202 committed:
[feat] closet: ClosetErrorCode에 NOT_FOUND 에러코드 추가

- CLOSET_NOT_FOUND 404 에러코드 추가
- 옷장이 존재하지 않거나 삭제된 경우 사용
- 403 에러코드 제거: Spring Security가 자동 처리

zerone1202 committed:
[feat] closet: ClosetSuccessCode에 옷장 상세 조회 성공 코드 추가

- CLOSET_GET_OK 200 OK 성공 코드 추가

zerone1202 committed:
[feat] closet: ClosetQueryServiceImpl에 옷장 상세 조회 메서드 추가

- getCloset() 메서드 구현
- closetId로 옷장 조회
- 존재하지 않는 옷장 404 예외 처리
- 삭제된 옷장(isDeleted = true) 404 예외 처리
- @Transactional(readOnly = true)로 조회 성능 최적화
- Entity → DTO 변환 (ClosetGetResponse.from 사용)

zerone1202 committed:
[feat] closet: 옷장 상세 조회 API 구현

- GET /api/v1/closets/{closetId} 엔드포인트 추가
- @PathVariable로 closetId 추출
- 인증 필요 (Spring Security 자동 처리)
- ApiResponse로 응답 반환
- ClosetSuccessCode.CLOSET_GET_OK 사용
- 200 OK 응답

zerone1202 committed:
Merge branch 'dev' of https://github.com/Princess-Maker-s-OOT/OOT-Outfit-of-Today into feat/closet-22
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/service/query/ClosetQueryServiceImpl.java:
```diff
@@ -33,4 +36,17 @@ public Page<ClosetGetPublicResponse> getPublicClosets(
+
+        if (closet.isDeleted()) {
+            throw new ClosetException(ClosetErrorCode.CLOSET_NOT_FOUND);
+        }
```
오~ 이 방식 좋네요! 수정하도록 하겠습니다.

zerone1202 replied:
이 방식이 테스트 코드에서 예외 상황 분리 어려움이 있다고 하여서 다시 제 원래 코드로 수정하겠습니다.

cjn404 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/controller/ClosetController.java:
컨벤션대로 수정 부탁드려요~

zerone1202 committed:
[feat] closet: ClosetQueryServiceImpl 에 Optional.filter를 사용

- Optional.filter를 사용하여 삭제된 옷장 필터링 로직 간결화
- 기존에는 옷장 조회 후 삭제 여부를 별도로 확인했으나, Optional.filter를 활용하여 삭제된 옷장을 조회 단계에서 걸러내도록 변경
- 기존 코드: findById → isDeleted 체크 → 예외 처리
- 변경 코드: findById().filter(!isDeleted) → orElseThrow()
- Optional의 체이닝을 활용해 불필요한 if 문을 제거

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/controller/ClosetController.java:
```diff
@@ -40,4 +41,20 @@ public ResponseEntity<ApiPageResponse<ClosetGetPublicResponse>> getPublicClosets
+     */
+    @GetMapping("/{closetId}")
+    public ResponseEntity<ApiResponse<ClosetGetResponse>> getCloset(
+            @PathVariable Long closetId
```
이건 추후 리팩토링에서 한번 더 생각해보겠습니다~!

cjn404 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/service/query/ClosetQueryServiceImpl.java:
```diff
@@ -33,4 +36,17 @@ public Page<ClosetGetPublicResponse> getPublicClosets(
+
+        if (closet.isDeleted()) {
+            throw new ClosetException(ClosetErrorCode.CLOSET_NOT_FOUND);
+        }
```
Closet closet = closetRepository.findByIdAndIsDeletedFalse(closetId)
    .orElseThrow(() -> new ClosetException(ClosetErrorCode.CLOSET_NOT_FOUND));

이런 방법도 추천드립니다~~

Byeongsu-cmd reviewed (approved):
코드 확인했습니다! 굿굿!

zerone1202 committed:
[feat] closet: ClosetErrorCode 404 에러 코드 추가

- 기존 CLOSET_NOT_FOUND와 의미적으로 분리
- CLOSET_DELETED 404응답으로 삭제된 리소스에 대해 명확한 응답 제공

zerone1202 committed:
[feat] closet: Optional.filter 제거 및 삭제 여부 명시적 분기 처리로 다시 변경

- 삭제된 옷장과 존재하지 않는 옷장을 구분하여 예외 처리
- 예외 발생 원인 명확화 및 가독성 향상
- 테스트 및 디버깅 시 예외 흐름 추적 용이

AllSungho reviewed (approved):
고생하셨습니다!
